ui/ui-components/pages/source/[id]/edit.tsx

```diff
@@ -152,7 +152,7 @@ export default function Page(props) {
   // not every row returned is guaranteed to have the same columns
   const previewColumns = preview
     .map((row) => Object.keys(row))
-    .flat()
+    .reduce((acc, val) => acc.concat(val), [])
```

Comment:
Unrelated, but `.flat()` isn't supported in our ES version... this is a polyfill from https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/flat